01-data-structures/01-introduction-to-data-structures/screen/screen.rb

```diff
@@ -6,18 +6,24 @@ class Screen
   attr_accessor :matrix
 
   def initialize(width, height)
+    self.width = width
+    self.height = height
+    self.matrix = Array.new(height){Array.new(width)}
   end
 
   # Insert a Pixel at x, y
   def insert(pixel, x, y)
+    self.matrix[x][y] = pixel
   end
 
   def at(x, y)
+    inbounds(x,y) ? self.matrix[x][y] : nil
   end
 
   private
 
   def inbounds(x, y)
+    (x > 0 || y > 0)
```

Comment:
What if the `height` and `width` are both 100, and I try to get the item at 150, 145?